extensions/amp-base-carousel/1.0/scroller.js

```diff
@@ -191,23 +191,24 @@ function ScrollerWithRef(
 
   // Trigger render by setting the resting index to the current scroll state.
   const debouncedResetScrollReferencePoint = useMemo(() => {
-    // Use local window if possible.
-    const win = containerRef.current ? getWin(containerRef.current) : window;
-    return debounce(
-      win,
-      () => {
-        // Check if the resting index we are centered around is the same as where
-        // we stopped scrolling. If so, we do not need to move anything.
-        if (
-          currentIndex.current === null ||
-          currentIndex.current === restingIndex
-        ) {
-          return;
-        }
-        setRestingIndex(currentIndex.current);
-      },
-      RESET_SCROLL_REFERENCE_POINT_WAIT_MS
-    );
+    // if containerRef is not set yet or serverside rendering, don't debouce.
+    return containerRef.current
+      ? debounce(
+          getWin(containerRef.current),
+          () => {
+            // Check if the resting index we are centered around is the same as where
+            // we stopped scrolling. If so, we do not need to move anything.
+            if (
+              currentIndex.current === null ||
+              currentIndex.current === restingIndex
+            ) {
+              return;
+            }
+            setRestingIndex(currentIndex.current);
+          },
+          RESET_SCROLL_REFERENCE_POINT_WAIT_MS
+        )
+      : () => undefined;
   }, [restingIndex, setRestingIndex]);
```

Comment:
You need to account for `containerRef.current`/`win` to bust the memo when the node is mounted.